src/time.cpp

```diff
@@ -17,118 +17,167 @@
 
 #include "meep.hpp"
 
+#include <algorithm>
+#include <iterator>
+#include <map>
+
 using namespace std;
 
 namespace meep {
 
+namespace {
+
+constexpr size_t kMeepTimingStackSize = 10;
```

Comment:
I'm not sure it's worth bounding the size the timing stack.  If the timing stack is growing out of control, we probably have bigger problems.

Reply:
Agreed. Just preserving existing behavior.

Reply:
The point is that if the stack is growing out of control then we have a bug.   It would be better to have an assertion failure in that case than silently discarding things.

The old behavior was because we were too lazy to do resizing for non-`std::vector` arrays, I think.

Reply:
Updated to assert.